### DISCUSSION TITLE
Dark mode classes not working as expected based on the documentation

### DISCUSSION THREAD

**Discussion by jithureddy** [tailwindlabs/tailwindcss#8745] [Help]
Hi, 
When I apply dark theme related class names as per documentation, styles are not applied as we can see below 
<img width="1542" alt="image" src="https://user-images.githubusercontent.com/5911189/176613537-b0c5bda0-9b7a-479f-80ae-7651ac206aaa.png">

But if change the order from `dark:hover:bg-gray-600` to `hover:dark:bg-gray-600` it work as shown below
<img width="1573" alt="image" src="https://user-images.githubusercontent.com/5911189/176613161-ad8ba4cf-e014-4f61-b0cc-2613fabf5681.png">

Tailwind config looks like below 
<img width="565" alt="image" src="https://user-images.githubusercontent.com/5911189/176613959-0ae1dc95-1ebe-4e74-9716-ed5293f004ac.png">

Please help or suggest
Thanks

**Comment by adamwathan** [comment]
Hey can you share an actual reproduction? This seems to work as expected to me:

https://play.tailwindcss.com/gam9Hydc1r

**Reply by jithureddy** [reply]
@adamwathan Thanks. Yeah above seems working fine but what I am missing in below project
https://github.com/jithureddy/tailwind-darkmode-menu/

Can still reproduce the issue

<img width="1789" alt="image" src="https://user-images.githubusercontent.com/5911189/176673867-f6ce4cc6-364b-4c23-a363-126446445ebb.png">
